native/membrane_video_compositor/src/compositor.rs

```diff
@@ -60,14 +64,19 @@ impl InnerState {
         &mut self,
         transformation: Arc<dyn UntypedTransformation>,
     ) -> Result<(), CompositorError> {
-        if self.transformations.contains_key(transformation.name()) {
-            return Err(CompositorError::TransformationNameTaken(
-                transformation.name().to_string(),
+        if self
```

Comment:
Perhaps we could gt rid of these repeatable code if we implemented some private function `register` and called it e.g. as: ` self.register(transformation.registry_key(), PluginRegistryEntry::Transformation(transformation))`